### ISSUE TITLE
[feature] Linear Cross Entropy Loss

### ISSUE THREAD

**Issue by nathan-az** [NVIDIA-NeMo/Megatron-Bridge#4665]
### User problem

Logit materialisation is immensely expensive for long context. Multiple good LCE implementations exist to support this (Cross Cut Entropy, Liger, etc.). They also appear to support softmax capping out of the box for models like Gemma4. These are much more memory efficient than the manual method, and I expect should be faster as more work is done in the same kernel (although I haven't confirmed this).

These are supported by Automodel so this is a case where Automodel outperforms Bridge on memory usage.

I note there has been discussion about a fused CE being implemented in Megatron/TE for the last 6+ months but so far has only landed in dev, and has not been as widely tested.

### Desired outcome

Add support for a fused CE implementation and a flag to enable it in the more common models, perhaps via a uv extra.

### Affected area

area:model

### Urgency / use case

Important but not blocking

**Comment by pavelgein** [comment]
There are several requests in Megatron-LM repo (but both of them into dev-branch)
- https://github.com/NVIDIA/Megatron-LM/pull/2256 merged, only for Blackwell
- https://github.com/NVIDIA/Megatron-LM/pull/3345 for Hopper 

But both of them missing softcapping

**Comment by nathan-az** [comment]
The same is true of PyTorch's [new native support](https://docs.pytorch.org/docs/2.13/generated/torch.nn.LinearCrossEntropyLoss.html) for it.

Bizarre choice given that both Liger and Cut Cross Entropy do support softcapping which are much older references.

**Comment by HowardZorn** [comment]
This is also good news for the chunked LM head with loss, as it can significantly reduce the VRAM footprint when using a large vocabulary size.
